src/cpp/server/server_cc.cc

```diff
@@ -171,6 +171,9 @@ bool ServerInterface::BaseAsyncRequest::FinalizeResult(void** tag,
     return true;
   }
   context_->set_call(call_);
```

Comment:
Instead of adding new code to all callers of `set_call()`, how about just adding a `ServerMetricRecorder*` parameter to `set_call()`?  That way, we're guaranteed that we don't accidentally miss any callers.

Reply:
Done, though since ServerMetricRecorder is optional I added two parameters.
Craig said this part will soon be refactored and he does not care much and was fine with the current version, but I think the updated version is still better as we don't repeat the same snippet. Let me know.